clients/cpp-console/src/common/Main.cpp

```diff
@@ -55,40 +55,39 @@ int main(int argc, char** argv)
         return (int)agentConfig->LoadResult();
     }
 
+    std::shared_ptr<DialogManager> dialogManager;
+    string keystroke = "";
+
     // Wavfile path to send to Speech Service
     if (wavFilePath != "")
     {
-        DialogManager dialogManager(agentConfig, wavFilePath);
-        dialogManager.ListenFromFile();
-        log_t("Initialized with audio file. Enter 'x' to exit.");
-        string keystroke = "";
-        cin >> keystroke;
-        HandleKeystrokeOptions(dialogManager, keystroke);
+        dialogManager = make_shared<DialogManager>(agentConfig, wavFilePath);
+        dialogManager->ListenFromFile();
     }
     else
     {
-        DialogManager dialogManager(agentConfig);
+        dialogManager = make_shared<DialogManager>(agentConfig);
 
         // Activate keyword listening on start up if keyword model file exists
         if (agentConfig->KeywordRecognitionModel().length() > 0)
         {
-            dialogManager.SetKeywordActivationState(KeywordActivationState::Paused);
-            dialogManager.StartKws();
+            dialogManager->SetKeywordActivationState(KeywordActivationState::Paused);
+            dialogManager->StartKws();
         }
         else
         {
-            dialogManager.SetKeywordActivationState(KeywordActivationState::NotSupported);
+            dialogManager->SetKeywordActivationState(KeywordActivationState::NotSupported);
         }
 
         DeviceStatusIndicators::SetStatus(DeviceStatus::Ready);
 
-        DisplayKeystrokeOptions(dialogManager);
-        log_t("Initialized with audio file. Enter 'x' to exit.");
-        string keystroke = "";
-        cin >> keystroke;
-        HandleKeystrokeOptions(dialogManager, keystroke);
+        DisplayKeystrokeOptions(*dialogManager);
     }
 
+    log_t("Initialized with audio file. Enter 'x' to exit.");
```

Comment:
"Initialized with audio file" only refres to the case wavFilePath != "", yet you log it for both live mic input and WAV file input

Reply:
Updated.